src/main/java/org/datadog/jenkins/plugins/datadog/util/git/GitMetadataBuilderCallback.java

```diff
@@ -75,36 +87,39 @@ private String getDefaultBranch(Repository repository, String remoteName) throws
         return null;
     }
 
-    private static String getBranch(Repository repository) throws IOException {
+    private static Collection<String> getBranches(Repository repository) throws IOException {
+        List<String> matchingBranches = new ArrayList<>();
+
         String branch = repository.getBranch();
         if (!GitUtils.isValidCommitSha(branch)) {
-          return branch;
+            matchingBranches.add(branch);
+            return matchingBranches;
         }
 
         // A detached HEAD is checked out.
-        // Iterate over available refs to see if any of them points to the checked out commit.
+        // Iterate over available refs to see which point to the checked out commit (there may be multiple).
         for (Ref ref : repository.getRefDatabase().getRefs()) {
             String refName = ref.getName();
             if (Constants.HEAD.equals(refName) || GitUtils.isValidCommitSha(refName)) {
                 continue;
             }
             ObjectId refObjectId = ref.getObjectId();
             if (branch.equals(refObjectId.getName())) {
-                return refName;
+                matchingBranches.add(normalizeBranch(refName));
             }
         }
-        return null;
+        return matchingBranches;
     }
 
     public static final class Result implements Serializable {
         final String repoUrl;
         final String defaultBranch;
-        final String branch;
+        final @Nonnull Collection<String> branches;
```

Comment:
Why not use `List<String>` to have more convenient methods for random access (e.g `get(0)`)?

Reply:
Rewrote this part